### ISSUE TITLE
PERF401 should not apply to nested list comprehensions

### ISSUE THREAD

**Issue by knyazer** [astral-sh/ruff#11936]
### Description
PERF401 recommends to use list comprehensions even if it leads to nested list comprehensions. I would guess a lot of people agree that nested list comprehensions are a bad code practice, but I cannot find any sources, sadly.

### Minimal example

```py
# test.py
def func():
    mylist = []
    for iterable in [[1], [2], [3]]:
        mylist.append([el + 1 for el in iterable])
    return mylist
```
When you run ```ruff check --select PERF401 test.py```, you would get a recommendation to use a list comprehension.

While minimal example does not look this bad, here is a real world example:
### Real example
```py
times = [
    [transcription.captions[index].end for index in range(len(transcription.captions) - 1)]
    for transcription in [*transcriptionStorage.get(), new_transcription]
    if transcription.isValid()
]
```
and wow, this is awful!

### Possible solution
I'm not sure how ruff internals work, but I guess it should not be extremely hard to figure out that there is a list comprehension inside of the block where the PERF401 is detected.

### Other
```ruff --version # 0.4.5```

**Comment by charliermarsh** [comment]
I see the motivation but I feel like this is working as intended. The rule is looking for iterations that could be rewritten as comprehensions for micro-optimization purposes, and it's identifying one here. To take the other side of the argument, it would be surprising for `PERF401` to sometimes trigger and sometimes not depending on the target expression. (Candidly, I don't find the real example to be much harder to read, but I recognize that's just subjective).

It's kind of a niche rule -- I'd recommend just disabling it if you're not in a situation that demands it, and aren't happy with the suggestions.
